lynxkite-app/web/tests/lynxkite.ts

```diff
@@ -132,10 +129,16 @@ export class Workspace {
   async connectBoxes(sourceId: string, targetId: string) {
     const sourceHandle = this.getBoxHandle(sourceId, "right");
     const targetHandle = this.getBoxHandle(targetId, "left");
+    await expect(sourceHandle).toBeVisible();
+    await expect(targetHandle).toBeVisible();
```

Comment:
This line is the key. I'm not sure why we even see the target box without a handle, but if we don't wait for it to appear, sometimes the edge doesn't get created between them. (I can't reproduce this manually.)